app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/DateOfBirthController.scala

```diff
@@ -54,12 +53,22 @@ class DateOfBirthController @Inject()(
     with SessionBehaviour {
 
   def showDateOfBirthForm(): Action[AnyContent] = Action.async { implicit request =>
-    withSubscribingAgent { _ =>
+    withSubscribingAgent { agent =>
       withValidSession { (_, existingSession) =>
-        existingSession.dateOfBirth match {
-          case Some(dob) =>
-            Ok(html.date_of_birth(dateOfBirthForm.fill(dob)))
-          case None => Ok(html.date_of_birth(dateOfBirthForm))
+        (agent.authNino, existingSession.nino, existingSession.dateOfBirthFromCid) match {
```

Comment:
The logic in this method is now a little bit opaque.  The method is still called "showDateOfBirthForm", but in fact this method can now show a bunch of different things depending on state.  To make it clearer, can we either rename the method, add some scaladoc or explain the reasoning behind each case please?